lib/Conversion/TorchOnnxToTorch/OnnxLstmExpander.cpp

```diff
@@ -0,0 +1,524 @@
+#include "torch-mlir/Conversion/TorchOnnxToTorch/Patterns.h"
+#include "torch-mlir/Conversion/TorchOnnxToTorch/Utils.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchOps.h"
+#include "torch-mlir/Dialect/Torch/Utils/Utils.h"
+
+using namespace mlir;
+using namespace mlir::torch::Torch;
+namespace mlir::torch::onnx_c {
+
+Value createActivationByName(ImplicitLocOpBuilder &b, StringRef name,
+                             Value input) {
+  if (name == "Sigmoid")
+    return b.create<AtenSigmoidOp>(input.getType(), input);
+  if (name == "Tanh")
+    return b.create<AtenTanhOp>(input.getType(), input);
+  if (name == "Relu")
+    return b.create<AtenReluOp>(input.getType(), input);
+  llvm_unreachable("Unsupported activation function");
+}
+
+/**
+ * @struct LstmWeights
+ * @brief A structure to hold LSTM weights.
+ *
+ * Each W_ weight matrix should have shape [hidden_size, input_size].
+ * Each R_ weight matrix should have shape [hidden_size, hidden_size].
+ * Each bias vector should have shape [4 * hidden_size].
+ */
+struct LstmWeights {
+  Value W_i, W_o, W_f, W_c;
+  Value R_i, R_o, R_f, R_c;
+  Value Wb_i, Wb_o, Wb_f, Wb_c;
+  Value Rb_i, Rb_o, Rb_f, Rb_c;
+};
+struct LstmActivations {
+  std::string f;
+  std::string g;
+  std::string h;
+};
+
+struct LstmCellState {
+  Value H;
+  Value C;
+};
+/**
```

Comment:
Just change this to proceeded `//` instead of a block comment. Everything else is fine, we just avoid using block comments so commenting out large blocks is feasible with block comments for debugging.